rotkehlchen/tests/api/test_evm_transactions.py

```diff
@@ -94,3 +105,122 @@ def test_query_transactions(rotkehlchen_api_server: 'APIServer'):
 
     assert optimism_found is True
     assert ethereum_found is True
+
+
+def test_event_with_details(rotkehlchen_api_server: 'APIServer'):
+    """Checks that if some events have details this is handled correctly."""
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    db = rotki.data.db
+
+    transaction = EvmTransaction(
+        tx_hash=deserialize_evm_tx_hash('0x66e3d2686193e01a4fbf0f598872236402edfe2f4efad84c4f6cdc753b8c78e3'),  # noqa: E501
+        chain_id=ChainID.ETHEREUM,
+        timestamp=Timestamp(1672580821),
+        block_number=16383832,
+        from_address=string_to_evm_address('0x0D268FE4F4BB33d092F098147646275241668A08'),
+        to_address=string_to_evm_address('0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45'),
+        value=0,
+        gas=21000,
+        gas_price=1000000000,
+        gas_used=21000,
+        input_data=b'',
+        nonce=26,
+    )
+    event1 = HistoryBaseEntry(
+        event_identifier=transaction.tx_hash,
+        sequence_index=221,
+        timestamp=ts_sec_to_ms(transaction.timestamp),
+        location=Location.BLOCKCHAIN,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_SUSHI,
+        balance=Balance(amount=FVal(100)),
+    )
+    event2 = HistoryBaseEntry(
+        event_identifier=transaction.tx_hash,
+        sequence_index=222,
+        timestamp=ts_sec_to_ms(transaction.timestamp),
+        location=Location.BLOCKCHAIN,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_USDT,
+        balance=Balance(amount=FVal(98.2)),
+        extra_data={
+            'sub-swaps': [
+                {'amount_in': '100.0', 'amount_out': '0.084', 'from_asset': A_SUSHI.identifier, 'to_asset': A_ETH.identifier},  # noqa: E501
+                {'amount_in': '0.084', 'amount_out': '98.2', 'from_asset': A_ETH.identifier, 'to_asset': A_USDT.identifier},  # noqa: E501
+            ],
+            'some-internal-data': 'some data',  # this data shouldn't be returned to the frontend
```

Comment:
As I said there is some extra information about staking that is returned in the case of liquity to the frontend for the staking view